### DISCUSSION TITLE
Can ganache be used to run full privatenet?

### DISCUSSION THREAD

**Discussion by 0x416e746f6e** [ConsenSys-archive/ganache#4286] [Q&A] [answered]
As in:  use `ganache` (possibly) in conjunction with some other components (like CL clients, validator clients, and so on) and have fully functional (post-merge) privatenet?

It appears to me that the answer is "no", because ganache is just a simulator.  Simply would like to confirm my understanding from a knowledgeable source.

**Comment by davidmurdoch** [comment, marked as answer]
Hi, maintainer here. The answer is no. Ganache does not know how to communicate with other external layers or nodes.

It's not out of the realm of possibility that we add the ability to stimulate these layers, if enough people are looking for that, in the future though.

**Comment by mksh** [comment]
I see that Lighthouse people use Ganache in CI/CD of their validator client: https://github.com/sigp/lighthouse/blob/0fb58a680d6f0c9f0dc8beecf142186debff9a8d/scripts/local_testnet/README.md?plain=1#L20, example run here: https://github.com/sigp/lighthouse/actions/runs/4339856397/jobs/7577840740 . @davidmurdoch Would be grateful if it's possible to explain how that could work if Ganache doesn't support CL connection. Thanks in advance!
